### PR TITLE
Fix: ISSUE-5546: replyTo are not updated if the original message is edited 

### PR DESCRIPTION
**Pull request by Claire1817**
## Type of change

- [ ] Feature
- [X] Bugfix
- [ ] Technical
- [ ] Other :

## Content
#5546 

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt [hunk 1]
```diff
@@ -411,9 +426,57 @@ internal class TimelineChunk(
     private fun buildTimelineEvent(eventEntity: TimelineEventEntity) = timelineEventMapper.map(
             timelineEventEntity = eventEntity,
             buildReadReceipts = timelineSettings.buildReadReceipts
-    ).let {
+    ).let { timelineEvent ->
         // eventually enhance with ui echo?
-        (uiEchoManager?.decorateEventWithReactionUiEcho(it) ?: it)
+        uiEchoManager?.decorateEventWithReactionUiEcho(timelineEvent)?.let { timelineEventWithEcho ->
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt:431]: This code is a bit hard to read...

Comment (Claire1817) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt:431]: i can create a val with the result if you prefer ?

Comment (ganfra) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt:431]: Maybe we can create an enhance delegation method, so we could add more in the future too

**Comment by ganfra** [comment]
We should take back the work on this. Will do next week.

**Comment by giomfo** [comment]
@ganfra can you please provide an updated ETA for this PR?

**Comment by ganfra** [comment]
I don't know when I'll have time to go back on it. Maybe next week as I have the Maintainer role.

**Comment by giomfo** [comment]
thx @ganfra to have updated this PR
@bmarty I let you handle the review process by assigning the right persons, in order to be able to merge it

**Review by bmarty — APPROVED** [review]
LGTM. To be safer, will be merged after the next RC.

**Comment by bmarty** [comment]
(I have merged develop into this branch, to trigger a CI build with the latest develop)

**Comment by bmarty** [comment]
The 2 use cases are still working as expected after a painful rebase:

### Redaction

- send a message
- send a reply to it
- delete the first message

### Edition

- send a message
- send a reply to it
- edit the first message

image

**Review by ganfra — APPROVED** [review]
LGTM thanks for the rebase

**Pull request merged**